### ISSUE TITLE
"Unable to obtain query #1" in CSV and XML cache

### ISSUE THREAD

**Issue by christabone** [jfear/sramongo#8]
It appears as if SRA can sometimes return an XML or CSV query with the following data:

```
<?xml version="1.0" encoding="UTF-8" ?>
<!DOCTYPE eEfetchResult PUBLIC "-//NLM//DTD efetch 20131226//EN" "https://eutils.ncbi.nlm.nih.gov/eutils/dtd/20131226/efetch.dtd">
<eFetchResult>
	<ERROR>Unable to obtain query #1</ERROR>
</eFetchResult>
```

This bypasses the timeout check (as it's returning data) but since the query has failed, nothing is stored in the XML or CSV cache files.

**Comment by christabone** [comment]
@jfear -- Would it work to try and catch this error here?

https://github.com/christabone/sramongo/blob/bc3b21c0406818dc069481f0030a96124ab01ab7/sramongo/sra2mongo.py#L210-L221

```
                try:
                    xml_handle = Entrez.efetch(retmode='xml', retstart=start, **options)
                    xml_data = xml_handle.read()
                    cache.add_to_cache(start, 'xml', xml_data)
                    xml_handle.close()

                    if runinfo_retmode is not None:
                        runinfo_handle = Entrez.efetch(rettype="runinfo", retmode=runinfo_retmode,
                                                       retstart=start, **options)
                        runinfo_data = runinfo_handle.read()
                        cache.add_to_cache(start, 'csv', runinfo_data)
                        runinfo_handle.close()
```

**Comment by jfear** [comment]
Hmm, this is a new one to me. Good idea to add some logic to check for this. Probably can just open and search the XMLs for `ERROR`.

**Comment by christabone** [comment]
(Whoops, mis-click).

Would it be best to catch this at the query stage so the query itself can be re-run from a specific point? Still reading around the code.

**Comment by jfear** [comment]
Have you been able to run through without errors?? I am interested to see if SRA gives you more problems because you are off NIH network. 

I am thinking of adding a small parse step in that try, to check for the ERROR and make sure the XML tags are closed. If an exception is raised then just re-run a few times. I can try to work on it tomorrow.

**Comment by christabone** [comment]
I restarted the process after removing the affected CSV / XML files and it appears to be loading normally. It's running on a laptop in my office so I'll provide an update tomorrow.

**Comment by christabone** [comment]
Update -- https://github.com/jfear/sramongo/issues/10

**Comment by jfear** [comment]
Is that the full XML in your original post? Going to use to make a test case.

**Comment by christabone** [comment]
Yes, 99% sure. Unfortunately, the file is overwritten so I can't be positive.

**Comment by jfear** [comment]
Ok, think I have the patch now. Will merge in once my other tests are done.

**Comment by jfear** [comment]
I think all of this should be fixed in PR #14
